icaruscode/CRT/CRTUtils/CRTHitRecoAlg.cc

```diff
@@ -1008,19 +1117,31 @@ sbn::crt::CRTHit CRTHitRecoAlg::MakeSideHit(vector<art::Ptr<CRTData>> coinData,
     t1trigs.resize(std::distance(t1trigs.begin(), std::unique(t1trigs.begin(), t1trigs.end())));
     //    t1hit = std::reduce(t1trigs.begin(), t1trigs.end()) / (uint64_t)t1trigs.size();
     //    t1hit = std::accumulate(t1trigs.begin(), t1trigs.end()) / (uint64_t)t1trigs.size();
-
-    for(uint64_t const t : ttrigs){
-      if (region=="North" || region=="South") /*thit += t;*/ thit += t-uint64_t(200.*fPropDelay)-fSiPMtoFEBdelay;
-      else /*thit += t;*/thit += t-uint64_t(400.*fPropDelay)-fSiPMtoFEBdelay;
-      //thit += t;
-      //thit += t-uint64_t(400.*fPropDelay);
-      if (fVerbose) 
-	mf::LogInfo("CRTHitRecoAlg: ") << " Inside the loop: t: \t"<< t << "\t" << uint64_t(200.*fPropDelay) 
+    if(ttrigs.size()>10){
+      for(uint64_t const t : ttrigs){
+        thit_s = t - t%1'000'000'000;
+	if (region=="North" || region=="South") thit_ns += t%1'000'000'000-uint64_t(200.*fPropDelay)-fSiPMtoFEBdelay;
+	else thit_ns += t%1'000'000'000 -uint64_t(400.*fPropDelay)-fSiPMtoFEBdelay; 
+      } 
+      if(fVerbose)
+	mf::LogInfo("CRTHitRecoAlg: ")  << "Average: thit_ns / ttrigs.size = " << thit_ns << " / " << uint64_t(ttrigs.size()) << " = ";
+      thit_ns = thit_ns /uint64_t(ttrigs.size()); //average ns portion of timestamp of CRT data products in the CRT Hit 
+      thit = thit_s + thit_ns; //add averaged ns portion of timestamp to full s portion of unix timestamp
+       if(fVerbose)
+	mf::LogInfo("CRTHitRecoAlg: ") << "add on full second: thit = thit_s + thit_ns = " << thit_s << " + " << thit_ns << " = " << thit << "\n";
+	    
+    }
+    else{
+      for(uint64_t const t : ttrigs){
+        if (region=="North" || region=="South") /*thit += t;*/ thit += t-uint64_t(200.*fPropDelay)-fSiPMtoFEBdelay;
+        else /*thit += t;*/thit += t-uint64_t(400.*fPropDelay)-fSiPMtoFEBdelay;
```

Comment:
Can we set this into its own function, since it's repeated?
```cpp
int64_t CRTHitRecoAlg::RegionDelay(std::string const& region) const {
  return fSiPMtoFEBdelay + uint64_t(((region=="North" || region=="South")? 200.: 400)*fPropDelay);
}
```
to be used like:
```suggestion
        thit -= RegionDelay(region);
```

Reply:
added this function